java.util.ConcurrentModificationException when using 2 schedulers and 2 rest clients

### Describe the bug

## 1. Dependencies 
I am using Quarkus version 2.5.3.Final

`pom.xml`
```xml
<dependencies>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-rest-client-reactive</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-resteasy-reactive</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-scheduler</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-rest-client-reactive-jackson</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-arc</artifactId>
    </dependency>
    <dependency>
      <groupId>io.quarkus</groupId>
      <artifactId>quarkus-junit5</artifactId>
      <scope>test</scope>
    </dependency>
    <dependency>
      <groupId>io.rest-assured</groupId>
      <artifactId>rest-assured</artifactId>
      <scope>test</scope>
    </dependency>
  </dependencies>
```
## 2. Files

### 1.
```java
@ApplicationScoped
public class FirstScheduler {

    public static final Logger logger = LoggerFactory.getLogger(FirstScheduler.class);

    @Inject
    @RestClient
    FirstClient firstClient;

    @Scheduled(every = "5m")
    void reload() {
        logger.info("FirstScheduler");
    }
}
```
### 2.
```java
@ApplicationScoped
public class SecondScheduler {

    public static final Logger logger = LoggerFactory.getLogger(SecondScheduler.class);

    @Inject
    @RestClient
    SecondClient secondClient;

    @Scheduled(every = "5m")
    void reload() {
        logger.info("SecondScheduler");
    }
}
```

 ### 3.

```java
@Path("/first")
@RegisterRestClient(configKey = "FirstClient")
public interface FirstClient {

    @GET
    @Path("/")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Uni<Object> fetch();
}
```

### 4. 

```java
@Path("/second")
@RegisterRestClient(configKey = "SecondClient")
public interface SecondClient {

    @POST
    @Path("/")
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    Uni<List<Object>> fetch();
}
```


### Expected behavior

The application should boot without errors.

### Actual behavior

The application boots but an error is thrown.

```logs
2021-12-16 14:31:04,019 ERROR [io.qua.sch.run.SimpleScheduler] (executor-thread-0) Error occured while executing task for trigger IntervalTrigger [id=1_com.test.schedulers.SecondScheduler_ScheduledInvoker_reload_e52552f7a8d66bad0704dbc1bc5053b5a80c5548, interval=300000]: java.lang.RuntimeException: Error injecting com.test.clients.SecondClient com.test.schedulers.SecondScheduler.secondClient
        at com.test.schedulers.SecondScheduler_Bean.create(Unknown Source)
        at com.test.schedulers.SecondScheduler_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.ClientProxies.getApplicationScopedDelegate(ClientProxies.java:18)
        at com.test.schedulers.SecondScheduler_ClientProxy.arc$delegate(Unknown Source)
        at com.test.schedulers.SecondScheduler_ClientProxy.reload(Unknown Source)
        at com.test.schedulers.SecondScheduler_ScheduledInvoker_reload_e52552f7a8d66bad0704dbc1bc5053b5a80c5548.invokeBean(Unknown Source)
        at io.quarkus.arc.runtime.BeanInvoker.invoke(BeanInvoker.java:20)
        at io.quarkus.scheduler.runtime.SimpleScheduler$ScheduledTask$1.run(SimpleScheduler.java:274)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$13.runWith(VertxCoreRecorder.java:548)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.util.ConcurrentModificationException
        at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1225)
        at io.quarkus.restclient.config.RestClientsConfig.getClientConfig(RestClientsConfig.java:65)
        at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.clientConfigByClassName(RestClientCDIDelegateBuilder.java:340)
        at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.configureBaseUrl(RestClientCDIDelegateBuilder.java:299)
        at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.build(RestClientCDIDelegateBuilder.java:63)
        at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.build(RestClientCDIDelegateBuilder.java:59)
        at io.quarkus.rest.client.reactive.runtime.RestClientCDIDelegateBuilder.createDelegate(RestClientCDIDelegateBuilder.java:42)
        at io.quarkus.rest.client.reactive.runtime.RestClientReactiveCDIWrapperBase.<init>(RestClientReactiveCDIWrapperBase.java:16)
        at com.test.clients.SecondClient$$CDIWrapper.<init>(Unknown Source)
        at com.test.clients.SecondClient$$CDIWrapper_Subclass.<init>(Unknown Source)
        at com.test.clients.SecondClient$$CDIWrapper_Bean.create(Unknown Source)
        at com.test.clients.SecondClient$$CDIWrapper_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
        at com.test.clients.SecondClient$$CDIWrapper_Bean.get(Unknown Source)
        at com.test.clients.SecondClient$$CDIWrapper_Bean.get(Unknown Source)
        ... 21 more


```

### How to Reproduce?

Download the zip 
[scheduler-clients-test.zip](https://github.com/quarkusio/quarkus/files/7727427/scheduler-clients-test.zip)
and run it

### Output of `uname -a` or `ver`

Darwin macbookpacegrey 21.1.0 Darwin Kernel Version 21.1.0: Wed Oct 13 17:33:23 PDT 2021; root:xnu-8019.41.5~1/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "11.0.9.1" 2020-11-04

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.5.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1

### Additional information

The same behaviour exists from `2.5.1.Final` version

--- a/extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientsConfig.java
+++ b/extensions/resteasy-classic/rest-client-config/runtime/src/main/java/io/quarkus/restclient/config/RestClientsConfig.java
@@ -1,10 +1,8 @@
 package io.quarkus.restclient.config;
 
-import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
-
-import org.wildfly.common.annotation.NotNull;
+import java.util.concurrent.ConcurrentHashMap;
 
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
@@ -29,7 +27,7 @@ public class RestClientsConfig {
     // The @Deprecated annotation prevents this field from being included in generated docs. We only want the `configKey` field
     // above to be included.
     @Deprecated
-    private final Map<String, RestClientConfig> configs = new HashMap<>();
+    private final Map<String, RestClientConfig> configs = new ConcurrentHashMap<>();
 
     /**
      * By default, REST Client Reactive uses text/plain content type for String values
@@ -65,7 +63,7 @@ public class RestClientsConfig {
         return configs.computeIfAbsent(configKey, RestClientConfig::load);
     }
 
-    public RestClientConfig getClientConfig(@NotNull Class<?> clientInterface) {
+    public RestClientConfig getClientConfig(Class<?> clientInterface) {
         return configs.computeIfAbsent(clientInterface.getName(), name -> RestClientConfig.load(clientInterface));
     }
 

